### ISSUE TITLE
redis `command` should accept arbitrary length of arguments not limited to four.

### ISSUE THREAD

**Issue by kairichard** [balabit/syslog-ng#2813]
# syslog-ng
## Version of syslog-ng

```
syslog-ng 3 (3.22.1)
Config version: 3.22
Installer-Version: 3.22.1
Revision: 3.22.1-2
Compile-Date: Jun 21 2019 12:50:04
Module-Directory: /usr/lib/syslog-ng/3.22
Module-Path: /usr/lib/syslog-ng/3.22
Include-Path: /usr/share/syslog-ng/include
Available-Modules: hook-commands,csvparser,appmodel,pseudofile,redis,json-plugin,confgen,afsocket,kvformat,afuser,mod-python,date,disk-buffer,xml,cef,basicfuncs,tags-parser,system-source,cryptofuncs,dbparser,afprog,linux-kmsg-format,affile,syslogformat,sdjournal
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: on
Enable-Systemd: on
```
# Issue

We want to add data to a redis stream with `XADD`, this redis command accepts an arbitrary length of fields, but the redis `command` only accepts four arguments which it too little for `XADD` and maybe even a few others. 

I also tried `command` with shoving all fields into the last argument which also does not work. 

## Configuration

```
destination d_redis {
    # I would love to use this but `command` only allows arguments
    # which for XADD is to little as we supply two key, value pairs `message_id` and `data`
    redis(
        host("redis-host")
        port(13451)
        auth("...")
        command("XADD", "instanceLogged", "*", "message_id $MESSAGE_UUID") # Here I want to add more `fields`
    );
};

```

**Comment by alltilla** [comment]
I agree, syslog-ng is not prepared to handle more than 3 fields. It should not be that hard to handle any given amount of fields. I will try to come up with a PR tomorrow.

**Comment by kairichard** [comment]
@alltilla wow, thank you so much for all the work that went into this. I am really looking forward to swapping out our python hack in syslog-ng with the proper redis implementation. 
Do you have a timeline on when this is going into a stable release?

**Comment by alltilla** [comment]
@kairichard You are welcome :)
This feature will be released in 3.23 version, which should come in the next 2 or 3 weeks.
